File: Lecture 001 Recursion Phod Denge/Factorial.cpp
#include<iostream>
using namespace std;

long long fact(int n){
    if(n==0) return 1;

    return n * fact(n-1);
}

int main(){
    int n;
    cin>>n;
    
    long long ans= fact(n);

    cout<<"factorial of "<<n<<" is: "<<ans<<endl;

    return 0;
}

File: Lecture 001 Recursion Phod Denge/Factorial.java
import java.util.Scanner;

public class Main {
	
	public static int fact(int n) {
         // base case 
		if(n==0) {
			return 1;
		}

        //tumhara kaam and trust on recursion 
		return (n * fact(n-1));
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int ans = fact(n);
	    System.out.println("Factorial of "+n+" is : "+ans);
	}
}